lib/classification_lifecycle.rb

```diff
@@ -36,14 +36,17 @@ def transact!(&block)
 
   def process_project_preference
     if should_create_project_preference?
-      existing_record = true
-      upp = UserProjectPreference.where(user: user, project: project).first_or_create do |up|
-        Project.increment_counter :classifiers_count, project.id
-        up.email_communication = user.project_email_communication
+      save_record = false
+      upp = UserProjectPreference.where(user: user, project: project).first_or_initialize do |up|
         up.preferences = {}
-        existing_record = false
+        save_record = true
+      end
+      if first_classifcation = upp.email_communication.nil?
+        save_record = true
+        Project.increment_counter :classifiers_count, project.id
+        upp.email_communication = user.project_email_communication
       end
-      upp.touch if existing_record
+      save_record ? upp.save! : upp.touch
```

Comment:
I wonder if you could just replace the whole `save_record` variable with `record.changed?`:

```ruby
      upp = UserProjectPreference.where(user: user, project: project).first_or_initialize do |up|
        up.preferences = {}
      end

      if first_classifcation = upp.email_communication.nil?
        Project.increment_counter :classifiers_count, project.id
        upp.email_communication = user.project_email_communication
      end

      upp.changed? ? upp.save! : upp.touch
```